api/python/provisioner/commands/validator/validator.py

```diff
@@ -255,3 +272,240 @@ class YumRepoDataValidator(DirValidator):
+        """
+        Validates the file content of the provided `path`.
+
+        Parameters
+        ----------
+        path: Path
+            File path for content validation
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValidationError
+            If validation is failed.
+        """
+        try:
+            logging.debug(f"File content type: '{self._content_type}'")
+            if self._content_type == ContentType.YAML:
+                content = utils.load_yaml(path)
+            elif self._content_type == ContentType.JSON:
+                content = utils.load_json(path)
+            else:
+                raise ValidationError(
+                            f"File content type '{self._content_type}'"
+                            " is not supported")
+        except Exception as e:
+            raise ValidationError(
+                            f"File content validation is failed: {e}") from e
+
+        logging.debug(f"File content: '{content}'")
+        try:
+            if isinstance(content, dict):
+                self._scheme.from_dict(content)
+            elif isinstance(content, list):
+                self._scheme.from_list(content)
+            else:
+                raise ValidationError("Unexpected top-level content type: "
+                                      f"'{type(content)}'")
+        except TypeError as e:
+            raise ValidationError(f"File content validation is failed: {e}")
+        else:
+            logging.info(f"File content validation is succeeded for '{path}'")
+
+    def __call__(self, *args, **kwargs):
+        """
+        Call-wrapper over self.validate method. It adds function-like behavior
+        for all class instances.
+
+        Parameters
+        ----------
+        args:
+            tuple of all positional arguments
+        kwargs:
+            dict with all keyword arguments
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValidationError
+            If validation is failed.
+        """
+        self.validate(*args, **kwargs)
+
+
+@attr.s(auto_attribs=True)
+class ReleaseInfoValidator(FileValidator):
+
+    """
+    Special alias for `RELEASE.INFO` file validator.
+
+    Attributes
+    ----------
+    _required: bool
+        if `True` validation raises an Exception if the file doesn't exist.
+        `True` by default
+    _content_type: ContentType
+        Content type of the `RELEASE.INFO` file.
+        `ContentType.YAML` is default value
+    """
+
+    _required: bool = attr.ib(
+        validator=attr.validators.instance_of(bool),
+        # NOTE: it is a difference in comparison of `FileValidator`
+        default=True
+    )
+    _content_type: ContentType = attr.ib(
+        validator=attr.validators.in_(ContentType),
+        default=ContentType.YAML
+    )
+
+    def __attrs_post_init__(self):
+        self._content_validator = ContentFileValidator(
```

Comment:
I can't do this. Because this class is inherited from `FileValidator`, where this attribute should be available at initialization time and  `__attrs_post_init__` method is not initialized for `FileValidator`. I can re-define this attribute in the class `ReleaseInfoValidator`. Do you wait exactly for this change from me?

Reply:
That attribute will be available but `attr` will set it itself:
```
  content_validator: Optional[Callable[[Path], None]] = attr.ib(
        init=False,
        validator=attr.validators.optional(attr.validators.is_callable()),
        default=None,
    )
```
It would just raise an error if someone tries to pass it explicitly.

Reply:
Improved